frame/support/src/traits/storage.rs

```diff
@@ -90,3 +92,35 @@ impl StorageInfoTrait for Tuple {
 pub trait PartialStorageInfoTrait {
 	fn partial_storage_info() -> Vec<StorageInfo>;
 }
+
+/// Allows a pallet to specify storage keys to whitelist during benchmarking.
+/// This means those keys will be excluded from the benchmarking performance
+/// calculation.
+pub trait WhitelistedStorageKeys {
+	/// Returns a [`Vec<TrackedStorageKey>`] indicating the storage keys that
+	/// should be whitelisted during benchmarking. This means that those keys
+	/// will be excluded from the benchmarking performance calculation.
+	fn whitelisted_storage_keys() -> Vec<TrackedStorageKey>;
+}
+
+#[cfg_attr(all(not(feature = "tuples-96"), not(feature = "tuples-128")), impl_for_tuples(64))]
+#[cfg_attr(all(feature = "tuples-96", not(feature = "tuples-128")), impl_for_tuples(96))]
+#[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
+impl WhitelistedStorageKeys for Tuple {
+	fn whitelisted_storage_keys() -> Vec<TrackedStorageKey> {
+		// de-duplicate the storage keys
+		// use BTreeMap so the resulting list of keys is unique
+		let mut combined_keys: BTreeMap<Vec<u8>, TrackedStorageKey> = BTreeMap::new();
```

Comment:
can't you use BTreeSet here? `TrackedStorageKey` should implement `PartialEq` as well

Reply:
BTreeSet doesn't actually exist / was removed at some point apparently. I tried

Reply:
Don't know what you are talking about :D 

https://paritytech.github.io/substrate/master/sp_std/collections/btree_set/struct.BTreeSet.html

Reply:
Ah no you know what it was it's this:
```
the trait `core::cmp::Ord` is not implemented for `TrackedStorageKey`
```

Reply:
Why not implement (not derive) `Ord` for `TrackedStorageKey`, where the implementation only looks at `.key`?